src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -25,28 +25,38 @@
 
 package com.fulcrumgenomics.fastq
 
-import java.io.Closeable
-
 import com.fulcrumgenomics.FgBioDef.{FgBioEnum, unreachable}
 import com.fulcrumgenomics.bam.api.{SamOrder, SamRecord, SamWriter}
 import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
 import com.fulcrumgenomics.commons.CommonsDef.{DirPath, FilePath, PathPrefix, PathToFastq}
-import com.fulcrumgenomics.commons.io.PathUtil
+import com.fulcrumgenomics.commons.async.{AsyncIterator, AsyncWriterPool}
+import com.fulcrumgenomics.commons.io.{PathUtil, Writer}
 import com.fulcrumgenomics.commons.util.{LazyLogging, Logger}
 import com.fulcrumgenomics.fastq.FastqDemultiplexer.{DemuxRecord, DemuxResult}
 import com.fulcrumgenomics.illumina.{Sample, SampleSheet}
 import com.fulcrumgenomics.sopt.{arg, clp}
 import com.fulcrumgenomics.umi.ConsensusTags
 import com.fulcrumgenomics.util.ReadStructure.SubRead
 import com.fulcrumgenomics.util.{ReadStructure, SampleBarcodeMetric, _}
+import enumeratum.EnumEntry
 import htsjdk.samtools.SAMFileHeader.SortOrder
 import htsjdk.samtools._
 import htsjdk.samtools.util.{Iso8601Date, SequenceUtil}
-import enumeratum.EnumEntry
 
 import scala.collection.immutable.IndexedSeq
 import scala.collection.mutable.ListBuffer
-import scala.concurrent.forkjoin.ForkJoinPool
+
+/** Controls how we read from the inputs in [[DemuxFastqs]].*/
+sealed trait DemuxFastqsAsyncReadingType extends EnumEntry
+case object DemuxFastqsAsyncReadingType extends FgBioEnum[DemuxFastqsAsyncReadingType] {
+  def values: IndexedSeq[DemuxFastqsAsyncReadingType] = findValues
+  /** Read in a synchronous fashion */
+  case object NoAsync extends DemuxFastqsAsyncReadingType
+  /** Read all the inputs in a single thread */
+  case object SingleThread extends DemuxFastqsAsyncReadingType
```

Comment:
Thinking about how this'll look on the command line I don't think this is super clear that it means a single _extra_ thread.  Maybe `SingleAsyncThread`?